cli/cmd/proxy/read/format_test.go

```diff
@@ -0,0 +1,220 @@
+			FullyQualifiedDomainName: "local_agent",
+			Endpoints:                []string{"192.168.79.187:8502"},
+			Type:                     "STATIC",
+			LastUpdated:              "2022-05-13T04:22:39.553Z",
+		},
+		{
+			Name:                     "client",
+			FullyQualifiedDomainName: "client.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul",
+			Endpoints:                []string{},
+			Type:                     "EDS",
+			LastUpdated:              "2022-06-09T00:39:12.948Z",
+		},
+		{
+			Name:                     "frontend",
+			FullyQualifiedDomainName: "frontend.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul",
+			Endpoints:                []string{},
+			Type:                     "EDS",
+			LastUpdated:              "2022-06-09T00:39:12.855Z",
+		},
+		{
+			Name:                     "local_app",
+			FullyQualifiedDomainName: "local_app",
+			Endpoints:                []string{"127.0.0.1:8080"},
+			Type:                     "STATIC",
+			LastUpdated:              "2022-05-13T04:22:39.655Z",
+		},
+		{
+			Name:                     "original-destination",
+			FullyQualifiedDomainName: "original-destination",
+			Endpoints:                []string{},
+			Type:                     "ORIGINAL_DST",
+			LastUpdated:              "2022-05-13T04:22:39.743Z",
+		},
+		{
+			Name:                     "server",
+			FullyQualifiedDomainName: "server.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul",
+			Endpoints:                []string{},
+			Type:                     "EDS",
+			LastUpdated:              "2022-06-09T00:39:12.754Z",
+		},
+	}
+
+	expectedHeaders := []string{"Name", "FQDN", "Endpoints", "Type", "Last Updated"}
+
+	actual := formatClusters(given)
+
+	require.Equal(t, expectedHeaders, actual.Headers)
+	require.Equal(t, len(given), len(actual.Rows))
+}
+
+func TestFormatEndpoints(t *testing.T) {
+	given := []Endpoint{
+		{
+			Address: "192.168.79.187:8502",
+			Cluster: "local_agent",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "127.0.0.1:8080",
+			Cluster: "local_app",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.31.201:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.47.235:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.71.254:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.63.120:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.18.110:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.52.101:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.65.131:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+	}
+
+	expectedHeaders := []string{"Address:Port", "Cluster", "Weight", "Status"}
+
+	actual := formatEndpoints(given)
+
+	require.Equal(t, expectedHeaders, actual.Headers)
+	require.Equal(t, len(given), len(actual.Rows))
```

Comment:
Given we are actually testing the formatting here, I think a strong assertion would be where we could actually compare the contents of actual vs what we think the output should be as a string. That would make the tests more robust, and also, we could actually test that it is formatted as expected.

Reply:
I like that. I'll need to run the render method and have it output into a buffer. I'll give that a quick shot. I think it's valuable and not too dissimilar from what I did for commands.

Reply:
Added!